ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/WSQueryEndpoint.java

```diff
@@ -288,13 +290,15 @@ private ServiceContext createServiceContext(final Principal principal) {
     // Creates a ServiceContext using the user's credentials, so the WS query topics are
     // accessed with the user permission context (defaults to KSQL service context)
 
-    if (!securityExtension.getUserContextProvider().isPresent()) {
+    if (principal == null || !securityExtension.getUserContextProvider().isPresent()) {
```

Comment:
Why would the principal here be null?

Reply:
If `null` is valid, then we should make this explicit by making the param `Optional`.